src/hugr/views/sibling_subgraph.rs

```diff
@@ -194,6 +194,51 @@ impl SiblingSubgraph {
         })
     }
 
+    /// Create a subgraph from a set of nodes.
+    ///
+    /// The incoming boundary is given by the set of edges with a source
+    /// not in nodes and a target in nodes. Conversely, the outgoing boundary
+    /// is given by the set of edges with a source in nodes and a target not
+    /// in nodes.
+    ///
+    /// The subgraph signature will be given by the types of the incoming and
+    /// outgoing edges ordered by the node order in `nodes` and within each node
+    /// by the port order.
+    pub fn try_from_nodes(
+        nodes: impl Into<Vec<Node>>,
+        hugr: &impl HugrView,
+    ) -> Result<Self, InvalidSubgraph> {
+        let nodes = nodes.into();
+        let nodes_set = nodes.iter().copied().collect::<HashSet<_>>();
+        let incoming_edges = nodes
+            .iter()
+            .flat_map(|&n| hugr.node_inputs(n).map(move |p| (n, p)));
+        let outgoing_edges = nodes
+            .iter()
+            .flat_map(|&n| hugr.node_outputs(n).map(move |p| (n, p)));
+        let inputs = incoming_edges
+            .filter(|&(n, p)| {
+                if !hugr.is_linked(n, p) {
+                    return false;
+                }
+                let (out_n, _) = hugr.linked_ports(n, p).exactly_one().ok().unwrap();
+                !nodes_set.contains(&out_n)
+            })
+            .map(|p| vec![p])
```

Comment:
I'm puzzled why we convert `p` to a singleton vector here, i.e. why is `IncomingPorts` a vector of vectors (not really pertinent to this PR though).